api-2.0/src/test/java/org/openmrs/module/attachments/AttachmentsServiceTest.java

```diff
@@ -0,0 +1,122 @@
+package org.openmrs.module.attachments;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.openmrs.Encounter;
+import org.openmrs.Obs;
+import org.openmrs.Concept;
+import org.openmrs.Patient;
+import org.openmrs.Visit;
+import org.openmrs.ConceptComplex;
+import org.openmrs.api.ConceptService;
+import org.openmrs.api.EncounterService;
+import org.openmrs.api.ObsService;
+import org.openmrs.api.PatientService;
+import org.openmrs.api.VisitService;
+import org.openmrs.module.attachments.obs.Attachment;
+import org.openmrs.module.attachments.obs.TestHelper;
+import org.openmrs.test.BaseModuleContextSensitiveTest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class AttachmentsServiceTest extends BaseModuleContextSensitiveTest {
+	
+	@Autowired
+	private AttachmentsService as;
+	
+	@Autowired
+	private TestHelper testHelper;
+	
+	@Autowired
+	@Qualifier("patientService")
+	private PatientService ps;
+	
+	@Autowired
+	@Qualifier("visitService")
+	private VisitService vs;
+	
+	@Autowired
+	@Qualifier("encounterService")
+	private EncounterService es;
+	
+	@Autowired
+	@Qualifier("conceptService")
+	private ConceptService cs;
+	
+	@Autowired
+	@Qualifier("obsService")
+	private ObsService os;
+	
+	@Test
+	public void getAttachments_shouldReturnEncounterAttachments() throws Exception {
+		
+		// Setup
+		List<Obs> complexObsList = testHelper.saveComplexObsForEncounter(2);
+		
+		Obs obs = complexObsList.get(0);
+		Patient patient = obs.getEncounter().getPatient();
+		Encounter encounter = obs.getEncounter();
+		Visit visit = encounter.getVisit();
+		
+		// Creating the Concept Complex Obs relevent to the attachments
+		ConceptComplex conceptComplex = cs.getConceptComplex(obs.getConcept().getConceptId());
+		// call the method in TestHelper.
+		Obs conceptComplexDefaultObs = testHelper.saveComplexObsForEncounterDefault(patient, conceptComplex, encounter);
+		
+		// saving some other obs during the same encounter
+		Obs otherObs = new Obs();
```

Comment:
i think creating an Obs should be done in a separate scoop as it doesn't impact directly the test, however i'm still not understanding it's use

Reply:
i mean enclose it in a bloc, an example is [Here](https://github.com/chinzou/openmrs-module-attachments/blob/0730834595b3bc1414a3cb93f73eaab207984a44/omod/src/test/java/org/openmrs/module/attachments/web/controller/AttachmentsControllerTest.java#L113)